src/node/node_state.h

```diff
@@ -452,17 +527,19 @@ namespace ccf
           quote_endorsements_client->fetch_endorsements();
           return;
         }
-
-        if (!((qi.format == QuoteFormat::oe_sgx_v1 &&
-               !qi.endorsements.empty()) ||
-              (qi.format != QuoteFormat::oe_sgx_v1 && qi.endorsements.empty())))
+        else // Non-SNP
         {
-          throw std::runtime_error(
-            "SGX quote generation should have already fetched endorsements");
-        }
+          if (!((quote_info.format == QuoteFormat::oe_sgx_v1 &&
```

Comment:
Separate, but that seems like it can go.

Reply:
Sure - for a future pass.